File: sert-2014/src/edu/wpi/first/wpilibj/templates/subsystems/ArmSubsystem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.wpi.first.wpilibj.templates.subsystems;
    
import edu.wpi.first.wpilibj.command.Subsystem;
import edu.wpi.first.wpilibj.Compressor;
import edu.wpi.first.wpilibj.DoubleSolenoid;
import edu.wpi.first.wpilibj.Relay;
import edu.wpi.first.wpilibj.Solenoid;
import edu.wpi.first.wpilibj.Timer;
import edu.wpi.first.wpilibj.templates.RobotMap;
import edu.wpi.first.wpilibj.templates.commands.Pressurize;
import edu.wpi.first.wpilibj.templates.commands.Pressurize;
/**
 *
 * @author SERT
 */

public class ArmSubsystem extends Subsystem {

    Compressor compressor;
    DoubleSolenoid lifter;

    private boolean isUp = true;
        
    public ArmSubsystem() {
        compressor = new Compressor(RobotMap.PRESSURE_SWITCH_CHANNEL, RobotMap.COMPRESSOR_CHANNEL);
        lifter = new DoubleSolenoid(RobotMap.SOLENOID_RAISE_CHANNEL, RobotMap.SOLENOID_LOWER_CHANNEL);
    }
    
    public boolean isUp() {
        return isUp;
    }
    
    public void raiseArm() {
        lifter.set(DoubleSolenoid.Value.kForward);
        isUp = true;
    }
    
    public void lowerArm() {
        lifter.set(DoubleSolenoid.Value.kReverse);
        isUp = false;
    }
    
    public void resetArmSolenoid() {
        lifter.set(DoubleSolenoid.Value.kOff);
    }
    
    public void startCompressor() {
        compressor.start();
    } 
    public void stopCompressor() {
        compressor.stop();
    }
    
    public void initDefaultCommand(){
        //setDefaultCommand(new Pressurize());
    }
}

File: sert-2014-C++/Subsystems/ArmSubsystem.cpp
#include "ArmSubsystem.h"

#include "../RobotMap.h"

ArmSubsystem::ArmSubsystem() :
b	Subsystem( "ArmSubsystem" )
{
	compressor = new Compressor(PRESSURE_SWITCH_CHANNEL, COMPRESSOR_CHANNEL);
	lifter = new DoubleSolenoid(SOLENOID_RAISE_CHANNEL, SOLENOID_LOWER_CHANNEL);
}

bool ArmSubsystem::IsUp() 
{
	return m_isUp;
}

void ArmSubsystem::RaiseArm() 
{
	lifter->Set( DoubleSolenoid::kForward );
	m_isUp = true;
}

void ArmSubsystem::LowerArm() 
{
	lifter->Set( DoubleSolenoid::kReverse );
	m_isUp = false;
}

void ArmSubsystem::ResetArmSolenoid() 
{
	lifter->Set( DoubleSolenoid::kOff );
}

void ArmSubsystem::StartCompressor() 
{
	compressor->Start();
} 
void ArmSubsystem::StopCompressor() 
{
	compressor->Stop();
}

void ArmSubsystem::InitDefaultCommand()
{
	//setDefaultCommand(new Pressurize());
}
